raft-tech/TANF-app
# Issue 1427: Remove usage of Cypress in the project

### Summary of Changes
Closes #1427 

Removes Cypress from the project entirely, with the exception of ignore files and historical Sprint documentation. 

### How to Test
```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
```

1. Check that the frontend builds and runs as expected
2. Open http://localhost:3000 and confirm that the page loads (no need to login)
3. Navigate to the Circle CI build output and confirm that no errors occur and Cypress was not run

#### Troubleshooting
If your build fails and the frontend won't start up, check the docker logs with `docker-compose logs tdp-frontend` from the `tdrs-frontend` directory. Look for the presence of this error:
```
Failed to compile.
tdp-frontend_1  | 
tdp-frontend_1  | Failed to load plugin 'import' declared in '.eslintrc.json �� eslint-config-react-app': Cannot find module '/home/node/app/node_modules/tsconfig-paths/node_modules/json5/lib/index.js'. Please verify that the package.json has a valid "main" entry
```
If found, this indicates stale packages in `node_modules` - follow these steps from your local machine to resolve:
1. Ensure `yarn` is installed and available on your path
2. Remove the yarn.lock file
```
rm yarn.lock
```
3. Remove node modules directory
```
rm -rf node_modules
```
4. Rebuild
```
docker-compose -f docker-compose.yml -f docker.compose.local.yml down
docker-compose -f docker-compose.yml -f docker.compose.local.yml up --build -d tdp-frontend
```

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

  - [x] CI builds no longer run Cypress tests or associated build steps

As Product Owner, @lfrohlich will decide if ACs are met.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*


### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*
    + [See the full Accessibility Assessment plan here.](https://github.com/HHS/TANF-app/blob/main/docs/Technical-Documentation/how-government-will-test-a11y.md)

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful deployment by assigning a 'Deploy with CircleCI - {env}' label*


### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*


### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*


### Deliverable 8: Context

> * Performance Standard(s): Code must be understandable and contextualized for the reviewers possess the knowledge and background necessary for analysis and constructive criticism to take place.

> * Acceptable Quality Level: Code submitted in the pull request has context.*
